hamilton/lifecycle/default.py

```diff
@@ -340,3 +346,118 @@ def run_before_node_execution(self, **future_kwargs: Any):
     def run_after_node_execution(self, **future_kwargs: Any):
         """Does nothing"""
         pass
+
+
+def check_instance(obj: Any, type_: Any) -> bool:
```

Comment:
Why are we not using this? https://github.com/DAGWorks-Inc/hamilton/blob/6bb3c6955ed35735cde4e2c9c67ef4019070151d/hamilton/htypes.py#L286